etcdserver/util.go

```diff
@@ -52,3 +52,34 @@ func numConnectedSince(transport rafthttp.Transporter, since time.Time, self typ
 	}
 	return connectedNum
 }
+
+// longestConnected chooses the member with longest active-since-time.
+// It returns false, if nothing is active.
+func longestConnected(tp rafthttp.Transporter, self types.ID, membs []types.ID) (types.ID, bool) {
```

Comment:
longestActive. Can you double check that tp.ActiveSince contains self information? or we do not have to add self as the arg.